### PR TITLE
Prepare v9.0.0 release (1st PS 0.15.0-compatible release)

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Backlinking to purescript/purescript#4244. Prepares project for first release that is compatible with PureScript v0.15.0.

:ping_pong: @thomashoneyman

:robot: This is an automated pull request to prepare the next release of this library. PR was created via the [Release.purs](https://github.com/JordanMartinez/purescript-ecosystem-update/blob/master/src/purescript/Command/Release.purs) file. Some of the following steps are already done; others should be performed by a human once the pull request is merged:

- [x] Updated bower dependencies to 0.15.0-compatible versions
- [x] ci.yml: Node already set to 14.
- [x] ci.yml: Spago tests uncommented.
- [x] Updated changelog
- [ ] Publish a GitHub [release](https://github.com/purescript-contrib/purescript-parsing/releases/new?tag=v9.0.0&title=v9.0.0&body=Breaking%20changes%3A%0A-%20Update%20project%20and%20deps%20to%20PureScript%20v0.15.0%20%28%23160%20by%20%40JordanMartinez%29%0A-%20Drop%20deprecated%20%60MonadZero%60%20instance%20%28%23160%20by%20%40JordanMartinez%29%0A-%20New%20optimized%20internals.%20%60ParserT%60%20now%20has%20a%20more%20efficient%20representation%2C%0A%20%20resulting%20in%20%28up%20to%29%2020x%20performance%20improvement.%20In%20addition%20to%20the%20performance%2C%0A%20%20all%20parser%20execution%20is%20always%20stack-safe%2C%20even%20monadically%2C%20obviating%20the%20need%0A%20%20to%20run%20parsers%20with%20%60Trampoline%60%20as%20the%20base%20Monad%20or%20to%20explicitly%20use%20%60MonadRec%60.%0A%0A%20%20Code%20that%20was%20parametric%20over%20the%20underlying%20Monad%20no%20longer%20needs%20to%20propagate%20a%0A%20%20Monad%20constraint.%0A%0A%20%20Code%20that%20constructs%20parsers%20via%20the%20underlying%20representation%20will%20need%20to%20be%20updated%2C%0A%20%20but%20otherwise%20the%20interface%20is%20unchanged%20and%20parsers%20should%20just%20enjoy%20the%20speed%20boost.%0A%0A%20%20%28%23154%20by%20%40natefaubion%29%0A-%20Make%20%60%3C%3F%3F%3E%60%20right-associative%20%28%23164%20by%20%40JordanMartinez%29%0A-%20Drop%20%60%3C%3F%3E%60%20and%20%60%3C~%3F%3E%60%20prec%20from%203%20to%204%20%28%23163%2C%20%23164%20by%20%40JordanMartinez%29%0A%0A%20%20%60%3C%7C%3E%60%20was%20made%20right%20associative.%20Decreasing%20these%20two%20operators%0A%20%20prevents%20a%20compiler%20error%20%28i.e.%20%60MixedAssociativityError%60%29%0A%20%20without%20causing%20issues%20with%20%60%3C%24%3E%60.%0A-%20Rename%20module%20prefix%20from%20%60Text.Parsing.Parser%60%20to%20%60Parsing%60%20%28%23169%20by%20%40jamesdbrock%29%0A-%20Replace%20the%20%60regex%60%20parser.%20%28%23170%20by%20%40jamesdbrock%29%0A-%20Reorganize%20Combinators%20for%20%23154%20%28%23182%20by%20%40jamesdbrock%29%0A-%20Add%20the%20%60index%60%20field%20to%20%60Position%60.%20%28%23171%20by%20%40jamesdbrock%29%0A-%20Move%20the%20parsers%0A%20%20*%20%60whiteSpace%60%0A%20%20*%20%60skipSpaces%60%0A%20%20*%20%60oneOf%60%0A%20%20*%20%60oneOfCodePoints%60%0A%20%20*%20%60noneOf%60%0A%20%20*%20%60noneOfCodePoints%60%0A%20%20from%20%60Parsing.String%60%20to%20%60Parsing.String.Basic%60.%20%28%23183%20by%20%40jamesdbrock%29%0A-%20Change%20MonadState%20instance%20%28%23187%20by%20jamesdbrock%29%0A%0A%20%20Users%20who%20stack%20a%20%60ParserT%60%20on%20a%20%60StateT%60%20base%20monad%20will%20call%20the%20%60MonadState%60%20members%20directly%20like%20%60get%60%20instead%20of%20needing%20to%20do%20%60lift%20%3C%3C%3C%20get%60.%0A%0A%20%20To%20get%20the%20%60ParserT%60%20internal%20state%2C%20call%20%60getParserT%60%20instead%20of%20%60get%60.%0A%0ANew%20features%3A%0A%0A-%20Add%20the%20%60anyTill%60%20primitive%20%60String%60%20combinator.%20%28%23186%20by%20%40jamesdbrock%29%0A-%20Add%20the%20%60Parsing.String.Replace%60%20module%2C%20copied%20from%0A%20%20https%3A%2F%2Fgithub.com%2Fjamesdbrock%2Fpurescript-parsing-replace%20%28%23188%20by%20%40jamesdbrock%29%0A%0ABugfixes%3A%0A%0A-%20Improve%20correctness%20and%20speed%20of%20%60number%60%20and%20%60intDecimal%60.%20%28%23189%20by%20%40jamesdbrock%29%0A%0AOther%20improvements%3A%0A%0A-%20Drop%20%60math%60%20dependency%3B%20update%20imports%20%28%23167%20by%20%40JordanMartinez%29
).
- [ ] Upload the release to Pursuit with `pulp publish`.

### CODE REVIEW

**Comment by JordanMartinez** [comment]
@thomashoneyman We might want to wait on this one so that @jamesdbrock can have one final go in the changes he's making before this library gets frozen.

**Comment by thomashoneyman** [comment]
Yep, I agree. @jamesdbrock, we pinged you a few times over the past few days on purescript.org/chat around the upcoming release of libraries for PureScript 0.15 but I'm not sure you saw the mentions.

We've commenced the PureScript 0.15 release and have released almost all the libraries in the core, contrib, web, and node organizations, but want to make sure you're still able to get changes in to this library that you want before PureScript 0.15. Of course, any upcoming breaking changes can always go into a subsequent breaking release for this library independent of the compiler.

Ideally we'd release this library for PureScript 0.15 in the next day or so, but if you need us to hold off for an urgent change then we can do that.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,16 @@ Notable changes to this project are documented in this file. The format is based
...
 
+Breaking changes:
+
+New features:
+
+Bugfixes:
+
+Other improvements:
+
+## [v9.0.0](https://github.com/purescript-contrib/purescript-parsing/releases/tag/v9.0.0) - 2022-04-27
```
Comment (thomashoneyman) [inline CHANGELOG.md:15]: We may need to remember to update this date if this release goes out later than today.

**Review by thomashoneyman — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (JordanMartinez) [reply CHANGELOG.md:15]: Ah, good point.

**Comment by jamesdbrock** [comment]
Hi I've merged everything, parsing is ready for release, thanks @JordanMartinez @thomashoneyman

**Comment by thomashoneyman** [comment]
Thanks for the heads up!

**Comment by thomashoneyman** [comment]
[v9.0.0](https://github.com/purescript-contrib/purescript-parsing/releases/tag/v9.0.0) is a huge release!

**Comment by JordanMartinez** [comment]
MAN! `parsing` was always just 'meh' but now it's like the swiss army knife it always wanted to be.

**Pull request merged**